node.py

```diff
@@ -333,14 +322,14 @@ def getdata(self, message):
 	def getblocks(self, message):
 		blkmeta = self.chaindb.locate(message.locator)
 		height = blkmeta.height
-		top_height = self.getheight()
+		top_height = self.chaindb.getheight()
 		end_height = height + 500
 		if end_height > top_height:
 			end_height = top_height
 
 		msg = msg_inv()
 		while height <= end_height:
-			hash = long(self.chaindb.height[str(height)])
+			hash = self.chaindb.getblockhash(height)
```

Comment:
I'm unsure what the intended behaviour was here, because previously `self.chaindb.height[str(height)]` could have returned a list of block hashes. For now, the new method defaults to the first hash.